test/form-builder/TextField.test.tsx

```diff
@@ -1,17 +1,83 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+/* eslint-disable @typescript-eslint/no-empty-function */
+
 import React from 'react';
 import TextField from '../../src/form-builder/TextField';
+import { waitFor } from '@testing-library/react';
+
+import { buildRenderForm, changeValue } from '../utils';
 
-import { buildRenderForm } from '../utils';
+const renderForm = buildRenderForm();
 
-const renderForm = buildRenderForm({ name: 'asdf' });
+const getInput = (container: HTMLElement) => {
+  const input = container.querySelector('va-text-input') as HTMLInputElement;
+  if (!input) throw new Error('No va-text-input found');
+  return input;
+};
 
 describe('Form Builder - TextField', () => {
   test('renders', () => {
-    const { getByLabelText } = renderForm(
+    const { container } = renderForm(
+      <TextField name="thing" label="The Thing" />
+    );
+    const input = getInput(container);
+    expect(input.getAttribute('label')).toEqual('The Thing');
+    expect(input.getAttribute('name')).toEqual('thing');
+  });
+
+  test('renders initial value', () => {
+    const rf = buildRenderForm({ thing: 'asdf' });
+    const { container } = rf(<TextField name="thing" label="The Thing" />);
+    const input = getInput(container);
+    expect(input.getAttribute('value')).toEqual('asdf');
+  });
+
+  test('renders the default "required" validation error message', async () => {
+    const { container, getFormProps } = renderForm(
+      <TextField name="thing" label="The Thing" required />
+    );
+    const input = getInput(container);
+    await waitFor(() => {
+      getFormProps().setFieldTouched('thing');
+    });
+    expect(input?.getAttribute('error')).toEqual('Please provide a response');
+  });
+
+  test('renders a custom "required" validation error message', async () => {
+    const { container, getFormProps } = renderForm(
+      <TextField
+        name="thing"
+        label="The Thing"
+        required="You need to fill this in, bub"
```

Comment:
![wolverine](https://media.giphy.com/media/12m3hgKuSuhClW/giphy.gif)

Reply:
With `label="The Thing"`, I realize now the required message should read `Fill this in or it's clobberin' time!"